### PR TITLE
Adds mood event checks in all_nighter quirk to stop runtimes from character setup menu

### PR DESCRIPTION
**Pull request by Diamond-74**
## About The Pull Request

Adds checks to the processing procs within the all_nighter quirk to make sure it doesn't try processing on a mob that doesn't have mood events

my frames

## Why It's Good For The Game

It runs faster without a bunch of runtimes.

## Changelog

:cl:
fix: character setup screen no longer runtimes with all_nighter quirk
/:cl:

### CODE REVIEW

**Comment by Diamond-74** [comment]
forgot to add description. gonna add that real quick.

**Comment by Mothblocks** [comment]
you also forgot the title.

**Comment by Diamond-74** [comment]
I looked at that mood config check and it's tied to the actual datum controller subsystem that processes quirks entirely. If there's a tie to the mob through the processor I can do that, but I didn't see any interactions between the processor and any actual mobs, just quirks.

**Comment by Mothblocks** [comment]
nvm that was the wrong prco

**Review by Mothblocks — APPROVED** [review]

File: code/datums/quirks/negative_quirks/all_nighter.dm [hunk 1]
```diff
@@ -99,9 +99,9 @@
 			happy_camper = FALSE
 
 	//adjusts the mood event accordingly
-	if(("all_nighter" in quirk_holder.mob_mood.mood_events) &&  happy_camper)
+	if(("all_nighter" in quirk_holder.mob_mood?.mood_events) &&  happy_camper)
```
Comment (Fikou) [inline code/datums/quirks/negative_quirks/all_nighter.dm:102]:
```suggestion
	if(("all_nighter" in quirk_holder.mob_mood?.mood_events) && happy_camper)
```

File: code/datums/quirks/negative_quirks/all_nighter.dm [hunk 2]
```diff
@@ -99,9 +99,9 @@
...
 		quirk_holder.clear_mood_event("all_nighter", /datum/mood_event/all_nighter)
-	if(!("all_nighter" in quirk_holder.mob_mood.mood_events) &&  !happy_camper)
+	if(!("all_nighter" in quirk_holder.mob_mood?.mood_events) &&  !happy_camper)
```
Comment (Fikou) [inline code/datums/quirks/negative_quirks/all_nighter.dm:104]:
```suggestion
	if(!("all_nighter" in quirk_holder.mob_mood?.mood_events) && !happy_camper)
```

**Comment by Diamond-74** [comment]
Thanks, didn't see the spacing errors

**Pull request merged**